### DISCUSSION TITLE
How to get RTKq interfaces ?

### DISCUSSION THREAD

**Discussion by Dovakeidy** [reduxjs/redux-toolkit#3615] [Q&A] [answered]
First of all, I really like RTK/RTKq and I thank you for the work you do! However, there's one thing I've always had trouble with, and that's access to the various RTK interfaces. 

For example, I've created a "useTable" hook to make it easier to manage the tables provided by tanstack table in combination with RTKq.

```ts
const { table } = useTable({
    columns,
    useQuery: ({ pagination }) => productApi.useReadManyProductQuery({
      pagination,
      sort: ['name:ASC'],
      populate: ['tags']
    })
  });
```

However, I've never found a way to access the return types of a hook generated by RTKq. In the example above, I'd like to have the return type of "useReadManyProductQuery" ( useXXXQuery  to be precise ) in order to properly type my "useTable" hook. 

Is there any way of accessing these interfaces?

Thanks in advance!

**Comment by EskiMojo14** [comment, marked as answer]
There are types exported from RTKQ's react module to help with this kind of thing, called TypedUseQueryHookResult and TypedUseMutationResult.

With that said, this usage seems to break the rules of hooks? To my knowledge you shouldn't ever pass a hook directly to another hook.

**Reply by Dovakeidy** [reply]
Thanks for the reply !!

About the use of the hook: the rules of hooks don't say you shouldn't do it and given that it works perfectly, without error, and that the use is practical I haven't found any reason not to do that.

Do you have any ?

**Reply by EskiMojo14** [reply]
it looks like the reason it's frowned upon is that it often means you're calling the hook conditionally, which *does* break the rules of hooks - as long as you are always calling the hook predictably, the order should be preserved and you'll be fine.

**Reply by Dovakeidy** [reply]
That's what I do, thanks @EskiMojo14 !
